WordPress/Classes/Services/WordPressComSyncService.swift

```diff
@@ -15,9 +15,39 @@ class WordPressComSyncService {
     ///     - onFailure: Closure to be executed upon failure.
     ///
     func syncWPCom(authToken: String, isJetpackLogin: Bool, onSuccess: @escaping (WPAccount) -> Void, onFailure: @escaping (Error) -> Void) {
+        syncAccountDetailsAndCreateAccount(authToken: authToken, onSuccess: { account in
+            self.syncOrAssociateBlogs(account: account, isJetpackLogin: isJetpackLogin, onSuccess: onSuccess, onFailure: onFailure)
+        }, onFailure: { error in
+            onFailure(error)
+        })
+    }
+
+    func syncAccountDetailsAndCreateAccount(authToken: String, onSuccess: @escaping (WPAccount) -> Void, onFailure: @escaping (Error) -> Void) {
```

Comment:
I wonder if we should have this new sync-and-create method inside `AccountService`. I found surprising to be passing a `RemoteUser` in a `.h` file, and it seems it's a first.

Seeing that this just calls `AccountServiceRemoteREST` and pass the result to another method in `AccountService` makes me think this belongs in there.